packages/Notification/Notification.jsx

```diff
@@ -31,32 +34,111 @@ const renderIcon = icon => <DecorativeIcon symbol={icon.symbol} variant={icon.co
  *
  * @version ./package.json
  */
-const Notification = ({ variant, children, ...rest }) => (
-  <Box {...safeRest(rest)} vertical={3} dangerouslyAddClassName={styles[variant]}>
-    <FlexGrid>
-      <FlexGrid.Row>
-        <FlexGrid.Col>
-          <Box inline between={3}>
-            {isImportant(variant) ? renderIcon(iconByVariant[variant]) : undefined}
-            <div
-              className={joinClassNames(
-                isImportant(variant) ? styles.withIcon : styles.withoutIcon
+class Notification extends React.Component {
+  state = {
+    dismissed: false,
+    contentWrapperHeight: undefined,
+  }
+
+  componentDidMount() {
+    if (this.props.dismissible) {
+      this.adjustContentHeight()
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.props.dismissible) {
+      this.adjustContentHeight()
+    }
+  }
+
+  adjustContentHeight = () => {
+    if (this.contentWrapper.offsetHeight !== this.state.contentWrapperHeight) {
+      this.setState({ contentWrapperHeight: this.contentWrapper.offsetHeight })
+    }
+  }
+
+  renderNotification() {
+    const { variant, dismissible, dismissibleA11yLabel, children, ...rest } = this.props
```

Comment:
We can add a warning when `dismissible` is defined and `dissmissibleA11yLabel` is undefined, similarly to this:

https://github.com/telus/tds-core/blob/be19a7e7de0790b74c109720edabbe9b29928c3c/packages/ChevronLink/ChevronLink.jsx#L35-L38